File: basic data structure code in c/sorting/selection_sort.c
#include<stdio.h>

void selectionsort(int[],int);

void main(){
	int n;
	printf("enter the size of array :");
	scanf("%d",&n);
	int arr[n];
	int i;
	for(i=0;i<n;i++){
		printf("ENTER THE %d ELEMENT : ",i+1);
		scanf("%d",&arr[i]);
	}

	printf("\n \t\t\t Your unsorted array is :   ");
	for(i=0;i<n;i++){
		printf("%d  ",arr[i]);
	}
	printf("\n");
	selectionsort(arr,n);
}

void selectionsort(int arr[],int n){
	int i,j,temp,min,k;
	for(i=0;i<n;i++){
		min=i;
		for(j=i+1;j<n;j++){
			if(arr[min]>arr[j]){
				min =j;
			}
		}
	//	here swapping are performed
			temp=arr[i];
			arr[i]=arr[min];
			arr[min]=temp;	
	}
	printf("\n \t\t\t   Your sorted array is :   ");
	for(k=0;k<n;k++){
		printf("%d  ",arr[k]);
	}
	
	printf("\n\n");
}

File: basic data structure using python/selection sort.py
#here instead of takin input from user we created a list having ome data element:
a=[3,8,5,2,4,1]
print(f"The data before sorting is : {a}")
print("sorting initiated...........")

#function is defined here
def selectionsort(a):
    n=len(a)
    for i in range(n):
        min=i
        for j in range(i+1, n):
            if(a[min]>a[j]):
                min=j

        a[i],a[min]=a[min],a[i]     #swapping done here


print("\n")
#function is called here.
selectionsort(a)
print(f"The sorted data is : {a}")
